code/modules/mob/living/carbon/xenomorph/castes/shrike/abilities_shrike.dm

```diff
@@ -344,44 +344,117 @@
 	succeed_activate()
 	add_cooldown()
-	var/obj/item/explosive/grenade/gravity/nade = new(T)
-	nade.throw_at(A, 5, 1, owner, TRUE)
-	nade.activate(owner)
-
-	owner.visible_message(span_warning("[owner] vomits up a roaring fleshy lump and throws it at [A]!"), span_warning("We vomit up a roaring fleshy lump and throws it at [A]!"))
-
-
-/obj/item/explosive/grenade/gravity
-	name = "gravity grenade"
-	desc = "A fleshy mass that seems way too heavy for its size. It seems to be vibrating."
-	arm_sound = 'sound/voice/predalien_roar.ogg'
-	greyscale_colors = "#3aaacc"
-	greyscale_config = /datum/greyscale_config/xenogrenade
-	det_time = 20
-
-/obj/item/explosive/grenade/gravity/prime()
-	new /obj/effect/overlay/temp/emp_pulse(loc)
-	playsound(loc, 'sound/effects/EMPulse.ogg', 50)
-	for(var/atom/movable/victim in view(3, loc))//yes this throws EVERYONE
-		if(victim.anchored)
-			continue
-		if(isliving(victim))
-			var/mob/living/livingtarget = victim
-			if(livingtarget.stat == DEAD)
+	addtimer(CALLBACK(owner, TYPE_PROC_REF(/mob, update_icons)), 1 SECONDS)
+	var/mob/living/carbon/xenomorph/xeno = owner
+	owner.icon_state = "[xeno.xeno_caste.caste_name][xeno.is_a_rouny ? " rouny" : ""] Screeching"
+	if(target) // Keybind use doesn't have a target
+		owner.face_atom(target)
+
+	playsound(owner, 'sound/effects/seedling_chargeup.ogg', 60)
+	ADD_TRAIT(owner, TRAIT_IMMOBILE, VORTEX_ABILITY_TRAIT)
+	if(!do_after(owner, VORTEX_PULL_WINDUP_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		for(var/atom/movable/victim in view(VORTEX_RANGE, owner.loc))
+			if(victim.anchored)
+				continue
+			if(isliving(victim))
+				var/mob/living_target = victim
+				if(living_target.stat == DEAD)
+					continue
+			victim.throw_at(owner, 3, 1, owner, FALSE)
+		return fail_activate()
+
+	for(var/turf/affected_tile in range(VORTEX_RANGE, owner.loc))
+		affected_tile.Shake(3, 3, 1 SECONDS)
+		for(var/i in affected_tile)
+			var/atom/movable/affected = i
+			if(!ishuman(affected) && !istype(affected, /obj/item) && !isdroid(affected))
+				affected.Shake(3, 3, 10)
 				continue
-		victim.throw_at(src, 5, 1, null, TRUE)
-	qdel(src)
+			if(ishuman(affected))
+				var/mob/living/carbon/human/H = affected
+				if(H.stat == DEAD)
+					continue
+				H.apply_effects(1,1)
+				H.adjust_stagger(2)
+				shake_camera(H, 2, 1)
+			var/throwlocation = affected.loc
+			for(var/x in 1 to 3)
+				throwlocation = get_step(throwlocation, owner.dir)
+			affected.throw_at(owner, 3, 1, owner, FALSE)
+
+	var/turf/targetturf = get_turf(owner)
+	targetturf = locate(targetturf.x + rand(1, 3), targetturf.y + rand(1, 3), targetturf.z)
+	if(do_after(owner, VORTEX_PUSH_WINDUP_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		for(var/atom/movable/victim in range(VORTEX_RANGE, owner.loc))
+			if(victim == owner)
+				continue
+			if(victim.anchored)
+				continue
+			if(isliving(victim))
+				var/mob/living_target = victim
+				if(living_target.stat == DEAD)
+					continue
+			victim.throw_at(targetturf, 3, 1, owner, FALSE)
```

Comment:
Are you copy pasting the same code here 3 times?
Why not just use a while loop?